analyzer/src/funTest/kotlin/GitRepoFunTest.kt

```diff
@@ -67,10 +68,12 @@ class GitRepoFunTest : StringSpec() {
         "Analyzer correctly reports VcsInfo for git-repo projects".config(enabled = !Ci.isAzureWindows) {
             val ortResult = Analyzer(DEFAULT_ANALYZER_CONFIGURATION).analyze(outputDir)
             val actualResult = ortResult.withResolvedScopes().toYaml()
-            val expectedResult = patchExpectedResult(
-                File("src/funTest/assets/projects/external/git-repo-expected-output.yml"),
-                revision = REPO_REV,
-                path = outputDir.invariantSeparatorsPath
+            val expectedResult = applyCustomSerialization(
```

Comment:
I have to admit, I'm a bit confused now about which format is used where. Maybe we could have a quick call with @oheger-bosch to clarify that?

Reply:
A call is fine with me; I should be available after 10.30 CET today.

Just to point out that the serialization problem is probably temporary. The custom serializer is currently needed because in order to access the dependency information, it has to be converted to the dependency tree format. In #4166 I am working on introducing an API to access this information independently on its representation; then no conversion will be required any more.

Reply:
As discussed in a call:
* In this case, we keep the comparison of strings in the new format, but the `applyCustomSerialization()` function was given a better name.
* In future, all expected result files are going to be converted to the new graph format; conversions are then no longer necessary.
* For the conversion, a helper-cli command will be added.